src/Microsoft.Identity.Web/Resource/AadIssuerValidator.cs

```diff
@@ -47,35 +47,36 @@ public static AadIssuerValidator GetIssuerValidator(string aadAuthority)
             if (string.IsNullOrEmpty(aadAuthority))
                 throw new ArgumentNullException(nameof(aadAuthority));
 
-            if (s_issuerValidators.TryGetValue(aadAuthority, out AadIssuerValidator aadIssuerValidator))
+            string authority;
+            try
             {
-                return aadIssuerValidator;
+                authority = new Uri(aadAuthority).Authority;
+            }
+            catch
+            {
+                authority = null;
             }
-            else
+
+            authority ??= new Uri(FallbackAuthority).Authority;
+
+
+            if (s_issuerValidators.TryGetValue(authority, out AadIssuerValidator aadIssuerValidator))
             {
-                // In the constructor, we hit the Azure AD issuer metadata endpoint and cache the aliases. The data is cached for 24 hrs.
-                var issuerMetadata = s_configManager.GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult();
-                string authorityHost;
-                try
-
-                {
-                    authorityHost = new Uri(aadAuthority).Authority;
-                }
-                catch
-                {
-                    authorityHost = null;
-                }
-
-                // Add issuer aliases of the chosen authority
-                string authority = authorityHost ?? new Uri(FallbackAuthority).Host;
-                var aliases = issuerMetadata.Metadata
-                    .Where(m => m.Aliases.Any(a => string.Equals(a, authority, StringComparison.OrdinalIgnoreCase)))
-                    .SelectMany(m => m.Aliases)
-                    .Append(authority) // For b2c scenarios, the alias will be the authorityHost itself
-                    .Distinct();
-                s_issuerValidators[authority] = new AadIssuerValidator(aliases);
-                return s_issuerValidators[authority];
+                return aadIssuerValidator;
             }
+
+            // In the constructor, we hit the Azure AD issuer metadata endpoint and cache the aliases. The data is cached for 24 hrs.
+            var issuerMetadata = s_configManager.GetConfigurationAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Doesn't this potentially make an http request? A blocking call here would take pretty long. What is the impact of making this method async?

Reply:
It does make a request. Making it an async is a breaking change, but it's an internal class... We should refactor this to use DI in general. I just wanted this PR/bug fix to be small and related to the cache issue.

Reply:
Block-Waiting for an http request to complete is kind of bad :) Good things there is a static cache...